Sources_v3/Controllers/MessageController/MessageController.swift

```diff
@@ -179,6 +222,59 @@ public extension _ChatMessageController {
             }
         }
     }
+    
+    /// Loads new messages from backend.
+    ///
+    /// - Parameters:
+    ///   - messageId: ID of the last fetched message. You will get messages `older` than the provided ID.
+    ///     In case no replies are fetched you will get the first `limit` number of replies.
+    ///   - limit: Limit for page size.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                 If request fails, the completion will be called with an error.
+    ///
+    func loadNextReplies(
+        after messageId: MessageId? = nil,
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        let lastMessageId = messageId ?? replies.last?.id
+    
+        messageUpdater.loadReplies(
+            cid: cid,
+            messageId: self.messageId,
+            pagination: MessagesPagination(pageSize: limit, parameter: lastMessageId.map { PaginationParameter.lessThan($0) })
+        ) { [weak self] error in
+            self?.callback { completion?(error) }
+        }
+    }
+    
+    /// Loads previous messages from backend.
+    ///
+    /// - Parameters:
+    ///   - messageId: ID of the current first message. You will get messages `newer` then the provided ID.
+    ///   - limit: Limit for page size.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                 If request fails, the completion will be called with an error.
+    ///
+    func loadPreviousReplies(
+        before messageId: MessageId? = nil,
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        guard let messageId = messageId ?? replies.first?.id else {
+            log.error(ClientError.MessageEmptyReplies().localizedDescription)
+            callback { completion?(ClientError.MessageEmptyReplies()) }
+            return
+        }
+    
+        messageUpdater.loadReplies(
+            cid: cid,
+            messageId: self.messageId,
+            pagination: MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
```

Comment:
I think there should be `lesttThen` here.

Reply:
Fixed in ae4e119 along with `ChannelController`pagination in 7f9ac07.